cmd/fossa/init.go

```diff
@@ -9,14 +10,34 @@ import (
 var initLogger = logging.MustGetLogger("init")
 
 func initCmd(c *cli.Context) {
-	conf, err := config.Initialize(c)
+	var err error
+	var conf config.CliConfig
+
+	conf, err = config.Initialize(c)
 	if err != nil {
 		initLogger.Fatalf("Could not load configuration: %s", err.Error())
 	}
-	doInit(conf)
+
+	conf.Modules, err = findModules(".")
+	if err != nil {
+		initLogger.Warningf("A non-fatal error occured during module discovery: %s", err.Error())
+	}
+
+	initLogger.Warningf("%v", conf)
 }
 
-// doInit calls DiscoverModules() on all available integrations and writes to
-func doInit(conf config.CliConfig) {
-	return
+// findModules calls DiscoverModules() on all available integrations and returns a new config state
+// to implement fail-open behavior, do not handle the err returned by this function
```

Comment:
I'm not sure what you mean by "fail-open" here.

Reply:
some integrations throw an error and we're still OK